### PR TITLE
Flickr: import primarily photo-wise rather than album-wise.

### PR DESCRIPTION
**Pull request by aboodman**
This way we get photos that aren't in any album.

Fixes: #2675

### CODE REVIEW

File: samples/js/flickr/slurp/src/main.js [hunk 1]
```diff
@@ -58,44 +62,72 @@ async function main(): Promise<void> {
...
+  return result;
+}
+
+async function getPhotosets(): Promise<any> {
+  return await flickr.callApi('flickr.photosets.getList');
 }
 
-function getPhotosets(flickr: Flickr): Promise<*> {
-  return flickr.callApi('flickr.photosets.getList').then(v => v.photosets.photoset);
+function updateStatus() {
```
Comment (arv) [inline samples/js/flickr/slurp/src/main.js]: Why don't you pass in `gottenPhotos` and `totalPhotos`?

Another option is to wrap these in a class with local state.

File: samples/js/flickr/slurp/src/main.js [hunk 2]
```diff
@@ -58,44 +62,72 @@ async function main(): Promise<void> {
...
+  const results = [];
+  for (let i = 2; i <= p1.photos.pages; i++) {
+    results.push(getPhotoPage(userId, privacyFilter, perPage, i));
+  }
+  return [p1].concat(await Promise.all(results));
+}
+
+async function getPhotoPage(userId: string, privacyFilter: number, perPage: number, pageNumber: number): Promise<any> {
+  const result = await flickr.callApi('flickr.photos.search', {
+    user_id: userId,
```
Comment (arv) [inline samples/js/flickr/slurp/src/main.js]:
```js
'user_id': userId,
```

Comment (aboodman) [inline samples/js/flickr/slurp/src/main.js]: fixed

File: samples/js/flickr/slurp/src/main.js (hunk 1 above)
Comment (aboodman) [inline samples/js/flickr/slurp/src/main.js]: I'm not going to do that.

**Comment by aboodman** [comment]
ptal

File: samples/js/flickr/slurp/src/main.js [hunk 3]
```diff
@@ -111,13 +109,16 @@ async function getPhotos(userId: string, privacyFilter: number): Promise<Array<a
...
 }
 
-async function getPhotoPage(userId: string, privacyFilter: number, perPage: number, pageNumber: number): Promise<any> {
+async function getPhotoPage(userId: string, privacyFilter: number, perPage: number,
+    pageNumber: number): Promise<any> {
   const result = await flickr.callApi('flickr.photos.search', {
-    user_id: userId,
-    privacy_filter: String(privacyFilter),
-    per_page: String(perPage),
+    ['user_id']: userId,
```
Comment (arv) [inline samples/js/flickr/slurp/src/main.js]: ` ['user_id']: userId,` ->  `'user_id': userId,`

**Pull request merged**